chiron/mcmc.py

```diff
@@ -0,0 +1,351 @@
+        )
+
+        dynamics.run(
+            x0=state_variables.positions,
+            temperature=state_variables.temperature,
+            n_steps=self.n_steps,
+            stepsize=self.stepsize,
+            collision_rate=self.collision_rate,
+        )
+
+
+class MCMove(StateUpdateMove):
+    def __init__(self) -> None:
+        self.system: Optional[NeuralNetworkPotential] = None
+
+    def _initialize_system(self, state_variables: SimulationState):
+        if self.system is None:
+            self.system = self.NeuralNetworkPotential(state_variables)
+
+    def _check_state_compatiblity(
+        self,
+        old_state: SimulationState,
+        new_state: SimulationState,
+    ):
+        """
+        Check if the states are compatible.
+
+        Parameters
+        ----------
+        old_state : StateVariablesCollection
+            The state of the system before the move.
+        new_state : StateVariablesCollection
+            The state of the system after the move.
+
+        Raises
+        ------
+        ValueError
+            If the states are not compatible.
+        """
+        pass
+
+    def apply_move(self):
+        """
+        Apply a Monte Carlo move to the system.
+
+        This method should be overridden by subclasses to define specific types of moves.
+
+        Raises
+        ------
+        NotImplementedError
+            If the method is not implemented in subclasses.
+        """
+
+        raise NotImplementedError("apply_move() must be implemented in subclasses")
+
+    def compute_acceptance_probability(
+        self,
+        old_state: SimulationState,
+        new_state: SimulationState,
+    ):
+        """
+        Compute the acceptance probability for a move from an old state to a new state.
+
+        Parameters
+        ----------
+        old_state : object
+            The state of the system before the move.
+        new_state : object
+            The state of the system after the move.
+
+        Returns
+        -------
+        float
+            Acceptance probability as a float.
+        """
+        self._check_state_compatiblity(old_state, new_state)
+        old_system = self.system(old_state)
+        new_system = self.system(new_state)
+
+        energy_before_state_change = old_system.compute_energy(old_state.position)
+        enegy_after_state_change = new_system.compute_energy(new_state.position)
+        # Implement the logic to compute the acceptance probability
+        pass
+
+    def accept_or_reject(self, probability):
+        """
+        Decide whether to accept or reject the move based on the acceptance probability.
+
+        Parameters
+        ----------
+        probability : float
+            Acceptance probability.
+
+        Returns
+        -------
+        bool
+            Boolean indicating if the move is accepted.
+        """
+        import jax.numpy as jnp
+
+        return jnp.random.rand() < probability
```

Comment:
We will definitely need to revisit how we call the RNG since they are not stateful in JAX.  To call the RNG you need to  pass a key (which should change each time).  So I think this class is going to need the `StateUpdateMove` is going to need something where we the initial random number key, and some internal variable that has the current subkey (that would be used by say, the `accept_or_reject` function, which would also need to generate a new subkey and log that). 

So like, if we assume we have a variable like `self.current_subkey` in the class, this function could be like:

`
from jax import random

self.current_subkey = random.split(self.current_subkey) 
return random.uniform(self.current_subkey) < probability
`

I think at this point, probably fine to just switch this to be numpy's random number generator which will work with the current syntax.